beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -242,35 +269,45 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
         &self,
         block: RpcBlock<T::EthSpec>,
     ) -> Result<MaybeAvailableBlock<T::EthSpec>, AvailabilityCheckError> {
-        let (block_root, block, blobs) = block.deconstruct();
-        match blobs {
-            None => {
+        let (block_root, block, blobs, data_columns) = block.deconstruct();
+        match (blobs, data_columns) {
+            (None, None) => {
                 if self.blobs_required_for_block(&block) {
                     Ok(MaybeAvailableBlock::AvailabilityPending { block_root, block })
                 } else {
                     Ok(MaybeAvailableBlock::Available(AvailableBlock {
                         block_root,
                         block,
                         blobs: None,
+                        data_columns: None,
                     }))
                 }
             }
-            Some(blob_list) => {
-                let verified_blobs = if self.blobs_required_for_block(&block) {
-                    let kzg = self
-                        .kzg
-                        .as_ref()
-                        .ok_or(AvailabilityCheckError::KzgNotInitialized)?;
-                    verify_kzg_for_blob_list(blob_list.iter(), kzg)
-                        .map_err(AvailabilityCheckError::Kzg)?;
-                    Some(blob_list)
-                } else {
-                    None
-                };
+            (maybe_blob_list, maybe_data_column_list) => {
+                let (verified_blobs, verified_data_column) =
+                    if self.blobs_required_for_block(&block) {
```

Comment:
is there a similar `data_columns_required_for_blocks` check we'll need to make here in the future?

Reply:
They are the same condition - `blobs_required_for_block` checks for number of commitments in the block - if there are commitments, there are blobs and we should verify data columns if they exist.